sound/soc/sof/ops.h

```diff
@@ -193,6 +193,15 @@ static inline int snd_sof_dsp_set_clk(struct snd_sof_dev *sdev, u32 freq)
 	return 0;
 }
 
+static inline int snd_sof_dsp_d0ix_update(struct snd_sof_dev *sdev,
+					  bool d0i3_enable)
```

Comment:
that doesn't seem to align with @lbetlej initial proposal. we wanted to have an IPC where you can set the state, not just a boolean.

Reply:
are you talking about PM_STATE_SET IPC between SW and FW? I haven't implemented IPC part yet(in the Todo list), please see inline:

```
	if (sdev->dsp_state == SOF_DSP_D0I0 && d0i3_enable) {
		dev_dbg(sdev->dev, "debug: entering to D0I3...\n");

		/* Todo: destroy unused pipelines/widgets in FW */

		/* do platform specific d0i3 entering task */
		ret = snd_sof_dsp_d0ix_update(sdev, d0i3_enable);

		/* sending d0i3_entry IPC */ //Pierre: I will add IPC sending PM_STATE_SET with D0I3 state here

		/* update dsp state */
		sdev->dsp_state = SOF_DSP_D0I3;

		return ret;
	}

	if (sdev->dsp_state == SOF_DSP_D0I3 && !d0i3_enable) {
		dev_dbg(sdev->dev, "debug: exiting from D0I3...\n");

		/* Todo: reload unused pipelines/widgets in FW */

		/* do platform specific d0i3 entering task */
		ret = snd_sof_dsp_d0ix_update(sdev, d0i3_enable);

		/* sending d0i3_exit IPC */ //Pierre: I will add IPC sending PM_STATE_SET with D0I0 state here

		/* update dsp state */
		sdev->dsp_state = SOF_DSP_D0I0;

		return ret;
	}
```